core/drivers/clk/clk-stm32-core.h

```diff
@@ -127,9 +130,132 @@ unsigned long clk_stm32_get_rate_divider(int div_id, unsigned long prate);
 int clk_stm32_parse_fdt_by_name(const void *fdt, int node, const char *name,
 				uint32_t *tab, uint32_t *nb);
 
+size_t clk_stm32_mux_get_parent(struct clk *clk);
+TEE_Result clk_stm32_mux_set_parent(struct clk *clk, size_t pidx);
+
+unsigned long clk_stm32_divider_get_rate(struct clk *clk,
+					 unsigned long parent_rate);
+
+TEE_Result clk_stm32_divider_set_rate(struct clk *clk,
+				      unsigned long rate,
+				      unsigned long parent_rate);
+
+size_t clk_stm32_composite_get_parent(struct clk *clk);
+TEE_Result clk_stm32_composite_set_parent(struct clk *clk, size_t pidx);
+unsigned long clk_stm32_composite_get_rate(struct clk *clk,
+					   unsigned long parent_rate);
+TEE_Result clk_stm32_composite_set_rate(struct clk *clk, unsigned long rate,
+					unsigned long parent_rate);
+TEE_Result clk_stm32_composite_gate_enable(struct clk *clk);
+void clk_stm32_composite_gate_disable(struct clk *clk);
+
+TEE_Result clk_stm32_set_parent_by_index(struct clk *clk, size_t pidx);
+
+unsigned long fixed_factor_get_rate(struct clk *clk, unsigned long parent_rate);
+
+extern const struct clk_ops clk_fixed_factor_ops;
+extern const struct clk_ops clk_fixed_clk_ops;
+extern const struct clk_ops clk_stm32_gate_ops;
+extern const struct clk_ops clk_stm32_gate_ready_ops;
+extern const struct clk_ops clk_stm32_divider_ops;
+extern const struct clk_ops clk_stm32_mux_ops;
+extern const struct clk_ops clk_stm32_composite_ops;
+
+#define PARENT(x...) { x }
+
+#define STM32_FIXED_RATE(_name, _rate)\
+	struct clk _name = {\
+		.ops	= &clk_fixed_clk_ops,\
+		.priv		= &(struct clk_fixed_rate_cfg) {\
```

Comment:
indentation here and in next macro definitions is a bit fuzzy.
Can you make is more consistent.
I would suggestion to not use tabulations but a single space char instead:
```c
#define ne STM32_FIXED_RATE(_name, _rate)\
	struct clk _name = {\
		.ops = &clk_fixed_clk_ops,\
		.priv = &(struct clk_fixed_rate_cfg) {\
			.rate = (_rate),\
		},\
		.name = #_name,\
		.flags = 0,\
		.num_parents = 0,\
	}
```

etc..